client/components/review.js

```diff
@@ -11,9 +11,99 @@ import { Markdown } from '@ukhomeoffice/asl-components';
 
 import ErrorBoundary from './error-boundary';
 import classnames from 'classnames';
+import isEqual from 'lodash/isEqual'; // for deep comparison
 
 class Review extends React.Component {
 
+hasDatabaseChange() {
+    const normalizeValue = value => {
+        if (value === null || value === undefined || (Array.isArray(value) && value.length === 0)) {
+            // Normalize `null`, `undefined`, and empty arrays to a consistent empty state
+            return null;
+        }
+        if (Array.isArray(value)) {
+            // Sort arrays to ensure order doesn't affect comparison
+            return value.sort();
+        }
+        if (typeof value === 'object' && value !== null) {
+            // Ensure consistent ordering of object keys for generic objects
+            return JSON.stringify(value, Object.keys(value).sort());
+        }
+        return value; // Return all other values as-is
+    };
+
+    const normalizeDuration = value => {
+        if (value && typeof value === 'object' && ('years' in value || 'months' in value)) {
+            // Normalize duration objects specifically
+            return {
+                years: value.years !== undefined && value.years !== null ? value.years : 5, // Default to 5 if undefined or null
+                months: value.months !== undefined && value.months !== null ? value.months : 0 // Default to 0 if undefined or null
+            };
+        }
+        return { years: 5, months: 0 }; // Default for completely missing duration
+    };
+
+    const { name, storedValue, currentValue, values } = this.props;
+
+    // Adjust storedValue for duration fields
+    const adjustedStoredValue =
+        name === 'duration' ? normalizeDuration(storedValue) : storedValue;
+
+    // Determine the actual current value for specific fields
+    const actualCurrentValue = currentValue !== undefined && currentValue !== null
+        ? currentValue
+        : values[name] !== undefined
+        ? values[name]
+        : null; // Fallback to `null` if no value is provided
```

Comment:
This is hard to read and can use the [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) to make it clearer
```suggestion
    const actualCurrentValue = currentValue ?? values[name] ?? null
```

Reply:
Jeff, I initially used the nullish operator for readability, but the explicit ternary check ensures falsy values like 0, "", and false are preserved, preventing unintended fallbacks, so I had to keep it this way.

Reply:
So does `??` - it only uses the RHS if the LHS is `null` or `undefined`. `||` has the behaviour you're describing, but that isn't what i'm suggesting